### DISCUSSION TITLE
Revert on passing public key on setAccountDataEncryptionKey in Account.sol

### DISCUSSION THREAD

**Discussion by nestorbonilla** [celo-org/social-connect#246] [Help (Q&A)]
**Description**
I’m trying to make from an account that I generated and added 5 CELO from the faucet, a DEK account, but when I pass the public key on setAccountDataEncryptionKey on the [Account.sol](https://explorer.celo.org/alfajores/address/0xed7f51A34B4e71fbE69B3091FcF879cD14bD73A9/write-proxy#address-tabs), it’s reverted.
[Loom](https://www.loom.com/share/0887602c5a614b9493e6efa34b829ee1) video for context.
**Question**
Which is the public key expected format?

**Comment by alecps** [comment]
Hey Nestor! It looks like the dek is normally wrapped by `ensureLeading0x` before it's passed to `setAccountDataEncryptionKey` (see example here https://github.com/celo-org/celo-monorepo/blob/80053fef19b4f543196b7a43215fd0644ba5db1b/packages/cli/src/commands/account/register-data-encryption-key.ts#L39). The Accounts contract will revert if the key length is less than 33, so that's another thing to check https://github.com/celo-org/celo-monorepo/blob/b5f69b76652f177f8d12ee083dabd82d61b7a823/packages/protocol/contracts/common/Accounts.sol#L245.
